From f40f9b5fac39dae0f446ffa2dc0a08ff4f50fdab Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Thu, 13 May 2010 19:27:01 -0400
Subject: [PATCH 197/390] markers-add-is-marker-enabled

Checks if a marker is enabled using a hash table lookup. API used by trace
control module.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@polymtl.ca>
---
 kernel/marker.c |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/kernel/marker.c b/kernel/marker.c
index ac8ec9f..7d11393 100644
--- a/kernel/marker.c
+++ b/kernel/marker.c
@@ -650,6 +650,25 @@ static void disable_marker(struct marker *elem)
 	 */
 }
 
+/*
+ * is_marker_enabled - Check if a marker is enabled
+ * @channel: channel name
+ * @name: marker name
+ *
+ * Returns 1 if the marker is enabled, 0 if disabled.
+ */
+int is_marker_enabled(const char *channel, const char *name)
+{
+	struct marker_entry *entry;
+
+	mutex_lock(&markers_mutex);
+	entry = get_marker(channel, name);
+	mutex_unlock(&markers_mutex);
+
+	return entry && !!entry->refcount;
+}
+EXPORT_SYMBOL_GPL(is_marker_enabled);
+
 /**
  * marker_update_probe_range - Update a probe range
  * @begin: beginning of the range
-- 
1.6.5.2

